src/dmd/target.d

```diff
@@ -1523,19 +1532,51 @@ struct Triple
         else
         {
             unknown(_os, "operating system");
-            return Target.OS.Freestanding;
+            return Target.OS.none;
         }
-        while (_os.length)
+
+        bool overflow;
+        auto major = parseNumber(_os, overflow);
+        if (overflow || major >= 255)
         {
-            if (!('0' < _os[0] && _os[0] < '9'))
-                break;
-            osMajor *= 10;
-            osMajor = cast(ubyte)((_os[0] - '0') + osMajor);
-            _os = _os[1 .. $];
+            error(Loc.initial, "OS version overflowed max of 254");
+            major = 255;
```

Comment:
Making it an errors would mean that [some valid triplets](https://issues.dlang.org/show_bug.cgi?id=22571) start being rejected.

Reply:
I've searched online and have not found a grammar on what is accepted and what isn't. No grammar was included with the documentation for this feature, either. No grammar on the wikipedia entry for it.

I'm leaning towards rejecting all triples that have no meaning for dmd. Otherwise, we risk:

1. if we eventually do recognize a certain sequence, we'll break existing usage that relied on it being ignored
2. setting user expectations that these ignored sequences have meaning